src/main/java/seedu/planner/model/ModulePlanner.java

```diff
@@ -451,6 +437,152 @@ private void sortAvailableModules(List<Module> modulesAvailable, UserProfile use
+                    && moduleDescription.get().getRequirement().toString().equals("Foundation")) {
+                count += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.FOUNDATION, new int[] {count});
+    }
+
+    /**
+     * Count the number of mathematics modules.
+     */
+    private void countMathematics() {
+        int count = 0;
+        for (Module m : getAllModulesTaken()) {
+            Optional<ModuleDescription> moduleDescription = getModuleDescription(m.getCode());
+            if (moduleDescription.isPresent()
+                    && moduleDescription.get().getRequirement().toString().equals("Mathematics")) {
+                count += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.MATHEMATICS, new int[]{count});
+    }
+
+    /**
+     * Count the number of science modules.
+     */
+    private void countScience() {
+        int count = 0;
+        for (Module m : getAllModulesTaken()) {
+            Optional<ModuleDescription> moduleDescription = getModuleDescription(m.getCode());
+            if (moduleDescription.isPresent()
+                    && moduleDescription.get().getRequirement().toString().equals("Science")) {
+                count += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.SCIENCE, new int[] {count});
+    }
+
+    /**
+     * Count the number of IT professionalism modules.
+     */
+    private void countItProfessionalism() {
+        int count = 0;
+        for (Module m : getAllModulesTaken()) {
+            Optional<ModuleDescription> moduleDescription = getModuleDescription(m.getCode());
+            if (moduleDescription.isPresent()
+                    && moduleDescription.get().getRequirement().toString().equals("IT professionalism")) {
+                count += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.IT_PROFESSIONALISM, new int[] {count});
+    }
+
+    /**
+     * Count the number of industrial experience modules.
+     */
+    private void countIndustrialExperienceRequirement() {
+        int count = 0;
+        for (Module m : getAllModulesTaken()) {
+            Optional<ModuleDescription> moduleDescription = getModuleDescription(m.getCode());
+            if (moduleDescription.isPresent()
+                    && moduleDescription.get().getRequirement()
+                    .toString().equals("Industrial Experience Requirement")) {
+                count += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT, new int[] {count});
+    }
+
+    /**
+     * Count the number of breadth and depth modules.
+     */
+    private void countBreadthAndDepth() {
+        List<FocusArea> userFocusAreas = new ArrayList<>(userProfile.getFocusAreas());
+        int[] count = new int[2];
+
+        for (Module m : getAllModulesTaken()) {
```

Comment:
The extracted method above might not work for `countBreathAndDepth`. But still good to refactor some parts here